PyInstaller/utils/hooks/qt.py

```diff
@@ -48,11 +48,11 @@ def __getattr__(self, name):
                 sys.stderr = sys.stdout
 
                 import json
-                from %s.QtCore import QLibraryInfo, QCoreApplication
+                from %s.QtCore import (QCoreApplication, QLibraryInfo)
 
                 # QLibraryInfo isn't always valid until a QCoreApplication is
                 # instantiated.
-                app = QCoreApplication([])
+                _ = QCoreApplication(sys.argv).instance().quit()
```

Comment:
Rewrite this to `app = QCoreApplication(sys.argv)`. Since the message loop is never entered, `quit` has no effect.

Reply:
# How PyQt5/Qt5 Exits Eventloop- or Main- Thread

I was mistaken in my apparent understanding that a call to QCoreApplication.exit() would  not only, kill the threads created to handle the event-loops, but also, kill the main-thread for the (core) application. My intent with the call to QCoreApplication.quit() (i.e., QCoreApplication.exit(0)) was to kill the main-thread of the (core) application.

The actual implementation is that a call to QCoreApplication.exit() kills the thread pertaining to the event-loop, while a call to QCoreApplication destructor kills the main-thread (see qcoreaplication_exec source code). This corresponds to your remark that QCoreApplication.quit() would be unnecessary.

I used the python sip module (thin layer that binds C++ and Python) to inquiry if the call to the QCoreApplication destructor must be made either by Python or by C++.

If the destructor call must be made by Python, then it will be implicitly made by Python-Sip bindings, while if it must be made by C++, then it must be explicitly made in the python script.

As shown in Fig. 1., The sip-dump suggests that the call will be implicitly made by the Python-Sip bindings, hence no code must be explicitly written to kill the main-thread. Therefore, upon garbage collection of the QCoreApplication instance, python will execute the sip-binding that calls C++ destructor. This will in turn kill the main-thread of the (core) application.

Figure 1. Output of Sip Dump States to be Destroyed by Python.

```
<PyQt5.QtCore.QCoreApplication object at 0x0000000002856708>
    Reference count: 2
    Address of wrapped object: 00000000028A2850
    Created by: Python
    To be destroyed by: Python
    Parent wrapper: NULL
    Next sibling wrapper: NULL
    Previous sibling wrapper: NULL
    First child wrapper: NULL
```

# Reference

[qcoreaplication_exec source code]( https://code.woboq.org/qt5/qtbase/src/corelib/kernel/qcoreapplication.cpp.html#_ZN16QCoreApplication4execEv )

Reply:
I agree.